Support Helm unittest snapshots (#19264)
pantsbuild/pants: docs/markdown/Helm/helm-overview.md

  - Add helm_chart target example
```

> If your workspace contains any Helm unit tests with snapshots (under an `__snapshot__` folder), `tailor` will identify those files are being part of your tests and will also create `helm_unittest_tests` and `resources` targets. See "Snapshot testing" below for more info.

Reviewer: ```suggestion
If your workspace contains any Helm unit tests with snapshots (under a `__snapshot__` folder), `tailor` will identify those files as test snapshots and will create `helm_unittest_tests` and `resources` targets. See "Snapshot testing" below for more info.
```

Reviewer: If I understand correctly, `tailor` will create `helm_unittest_tests` even without a `__snapshot__` folder? This reads as if the `helm_unittest_tests` target creation depends on their being a `__snapshots__` folder. But I think those are two separate bits of logic in tailor? Did this mean to say that Pants will infer a dependency from the `helm_unittests_tests` target on the associated `resources` target?

Author: Yeah, that's what I meant, unit tests and snapshots are identified by tailor independently and a dependency will be inferred from the former to the later in case snapshots are found.

Reviewer: So this needs some rephrasing...

> not affected by the source roots setting. When using `relocated_files`, the files will be relative
> to the value set in the `dest` field.

### Snapshot testing

> Using unit test snapshots is fully supported by Pants as test snapshots are just resources as shown in the previous section.

Reviewer: ```suggestion
Unit test snapshots are supported by Pants by wrapping the snapshots in resources targets, as shown in the previous section.
```

> However, since managing snapshots by hand is quite tedious, Pants gives some utilities to manage them in a simpler way. To automate the update of the snapshots use Pants' `generate-snapshots` goal which we can use to generate the snapshot files:

Reviewer: ```suggestion
Since managing snapshots by hand is quite tedious, Pants provides some utilities to manage them in a simpler way. To generate or update the snapshots, use Pants's `generate-snapshots` goal:
```

Reviewer: If the `s` at the end of a word indicates that it's plural, then the possessive is indeed `s'` (like "my friends' names"). But `Pants` is a singular noun ("Pants is a build system", not "Pants are a build system"), so the possessive is `Pants's`. 

If we were talking about trousers, then it is plural ("my Pants are tight", not "my Pants is tight") you would be right, the possessive would be `Pants'`. 

English is a very stupid language...

Author: It is indeed! 😂

```
pants --filter-target-type='helm_unittest_test' generate-snapshots ::
```

> That would generate the test snapshots for those tests that require it with older snapshots being overwritten and newer ones being created.

Reviewer: ```suggestion
This will generate test snapshots for tests that require them, with older snapshots being overwritten by newer ones.
```

> If new `__snapshot__` folders have been created we recommend to run again the `tailor` goal and Pants would recognise those new folders and create `resource` targets as appropriate.

Reviewer: ```suggestion
If new `__snapshot__` folders are created we recommend running the `tailor` goal, so that Pants can detect these new folders and create `resource` targets as appropriate.
```
